ops/framework.py

```diff
@@ -758,7 +758,9 @@ def _reemit(self, single_event_path=None):
                 if custom_handler:
                     event_is_from_juju = isinstance(event, charm.HookEvent)
                     event_is_action = isinstance(event, charm.ActionEvent)
-                    if (event_is_from_juju or event_is_action) and 'hook' in self._juju_debug_at:
+                    if (event_is_from_juju or event_is_action) and any(
+                        x in self._juju_debug_at for x in ["all", "hook"]
```

Comment:
this feels like an inefficient way to do this, but it does seem correct. And theoretically _juju_debug_at should generally be small.
I do wonder if we want to change the parsing from:
```
        # Parse the env var once, which may be used multiple times later
        debug_at = os.environ.get('JUJU_DEBUG_AT')
        self._juju_debug_at = debug_at.split(',') if debug_at else ()
```

to
```
        # Parse the env var once, which may be used multiple times later
        debug_at = os.environ.get('JUJU_DEBUG_AT', '')
        self._juju_debug_at = set(x.strip() for x in debug_at.split(','))
```
and then you could do:
```
and self._juju_debug_at.intersection({'all', 'hook'})
```
```
$ python -m timeit -s "debug_at = 'a,b,c,d,e,f,g,all'.split(',')" "any(x in debug_at for x in ['all', 'hook'])"
1000000 loops, best of 5: 285 nsec per loop
$ python -m timeit -s "debug_at = set([a.strip() for a in 'a,b,c,d,e,f,g,all'.split(',')])" "debug_at.intersection({'all', 'any'})"
2000000 loops, best of 5: 101 nsec per loop
```

about 2x faster, but also only 100ns, so probably premature optimization.
Though interestingly, still better even in the single-object case:
```
$ python -m timeit -s "debug_at = 'all'.split(',')" "any(x in debug_at for x in ['all', 'hook'])"                      1000000 loops, best of 5: 226 nsec per loop
(3.8) jameinel@focal:~/dev/operator (master)
$ python -m timeit -s "debug_at = set([a.strip() for a in 'all'.split(',')])" "debug_at.intersection({'all', 'any'})"  5000000 loops, best of 5: 96.1 nsec per loop
```

I guess I'd go more for whatever is the most understandable in code.